gli/io.py

```diff
@@ -1451,3 +1470,871 @@ def save_task_node_classification(name,
+                "key": "test_set"
+            }
+        }
+    """  # noqa: E501,E262  #pylint: disable=line-too-long
+    # Check the input arguments.
+    assert isinstance(description, str), \
+        "`description` must be a string."
+    _check_feature(feature)
+    task_type = "LinkPrediction"
+    task_str = "link_prediction"
+
+    task_dict = {
+        "description": description,
+        "type": task_type,
+        "feature": feature,
+    }
+    data_dict = {
+        "train_set": np.array(train_set),
+        "val_set": np.array(val_set),
+        "test_set": np.array(test_set)
+    }
+    if val_neg is not None:
+        data_dict["val_neg"] = val_neg
+    if test_neg is not None:
+        data_dict["test_neg"] = test_neg
+    key_to_loc = save_data(f"{name}__task_{task_str}_{task_id}",
+                           save_dir=save_dir,
+                           **data_dict)
+    task_dict.update(key_to_loc)
+
+    # Save the task json file.
+    with open(os.path.join(save_dir, f"task_{task_str}_{task_id}.json"),
+              "w",
+              encoding="utf-8") as f:
+        json.dump(task_dict, f, indent=4)
+
+    return task_dict
+
+
+def save_task_time_dependent_link_prediction(name,
+                                             description,
+                                             feature,
+                                             time,
+                                             train_time_window=None,
+                                             val_time_window=None,
+                                             test_time_window=None,
+                                             val_neg=None,
+                                             test_neg=None,
+                                             task_id=1,
+                                             save_dir="."):
+    """Save the time dependent link prediction task information into task json and data files.
+
+    :param name: The name of the dataset.
+    :type name: str
+    :param description: The description of the task.
+    :type description: str
+    :param feature: The list of feature names to be used in the task. The
+        features could be node attributes, edge attributes, or graph
+        attributes. For homogeneous graphs, the feature names should be in the
+        format of "Node/{node_attr_name}", "Edge/{edge_attr_name}", or
+        "Graph/{graph_attr_name}". For heterogeneous graphs, the feature names
+        should be in the format of "Node/{node_type}/{node_attr_name}",
+        "Edge/{edge_type}/{edge_attr_name}", or
+        "Graph/{graph_type}/{graph_attr_name}". The node/edge/graph_attr_name,
+        and the node/edge/graph_type should be the ones declared in the
+        metadata.json file.
+    :type feature: list of str
+    :param time: The time attribute that indicates edge order in the format of "Edge/{time_type}"
```

Comment:
-> The time attribute that indicates the formation time or order of edges in the format of "Edge/{edge_attr_name}.